### DISCUSSION TITLE
Documento não localizado

### DISCUSSION THREAD

**Discussion by MikeThmz** [lucashpmelo/node-mde#11] [Q&A]
@lucashpmelo, primeiramente agradeço seu esforço de criação e disponibilzação deste componente.
Estou testando e avaliando e fiquei com uma dúvida. Numa busca com a função consultaUltNSU, eu obtive a resposta de acordo, porém ele dá um retorno de Nenuym documento localizado. Porém ele traz o ultNSU e maxNSU.
No meu entendimento, se existem NSUs, deveria trazer documentos. Onde estou errando no meu entendimento?

Obrigado novamente!!

`{
  data: {
    tpAmb: '1',
    verAplic: '1.6.2',
    cStat: '137',
    xMotivo: 'Nenhum documento localizado',
    dhResp: '2023-07-14T15:14:43-03:00',
    ultNSU: '000000000000050',
    maxNSU: '000000000000417',
    docZip: []
  },
  reqXml: '<?xml version="1.0" encoding="utf-8"?><soap12:Envelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:soap12="http://www.w3.org/2003/05/soap-envelope"><soap12:Body><nfeDistDFeInteresse xmlns="http://www.portalfiscal.inf.br/nfe/wsdl/NFeDistribuicaoDFe"><nfeDadosMsg><distDFeInt xmlns="http://www.portalfiscal.inf.br/nfe" versao="1.01"><tpAmb>1</tpAmb><cUFAutor>35</cUFAutor><CPF>XXXXXXXXXXX</CPF><distNSU><ultNSU>000000000000000</ultNSU></distNSU></distDFeInt></nfeDadosMsg></nfeDistDFeInteresse></soap12:Body></soap12:Envelope>',
  resXml: '<?xml version="1.0" encoding="utf-8"?><soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><soap:Body><nfeDistDFeInteresseResponse xmlns="http://www.portalfiscal.inf.br/nfe/wsdl/NFeDistribuicaoDFe"><nfeDistDFeInteresseResult><retDistDFeInt xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" versao="1.01" xmlns="http://www.portalfiscal.inf.br/nfe"><tpAmb>1</tpAmb><verAplic>1.6.2</verAplic><cStat>137</cStat><xMotivo>Nenhum documento localizado</xMotivo><dhResp>2023-07-14T15:14:43-03:00</dhResp><ultNSU>000000000000050</ultNSU><maxNSU>000000000000417</maxNSU></retDistDFeInt></nfeDistDFeInteresseResult></nfeDistDFeInteresseResponse></soap:Body></soap:Envelope>',
  status: 200
}`

**Comment by lucashpmelo** [comment]
Boa tarde @MikeThmz, o serviço de Distribuição de DF-e apenas gera NSU para notas ou eventos se o mesmo for utilizado dentro de um período de 60 dias, depois disso ele suspende a geração de novos NSUs até que seja feita uma nova consulta pelo ultimo NSU recebido.

Os documentos também só ficam disponíveis para consulta por um período limite de no máximo 90 dias, depois não é mais possível recuperar os XMLs por esse serviço.

Esse deve ser seu caso nesse momento, o serviço gerou NSU para esse autor até o numero 417, depois como não houve mais consultas a geração foi interrompida e os documentos passaram do período de 90 dias. Como agora você acabou de realizar uma consulta a Sefaz irá voltar a gerar NSUs para esse autor, então as próximas notas emitidas contra ele começarão a aparecer. Sua próxima consulta deve começar pelo NSU 000000000000417.

Recomendo a leitura da [Nota Técnica 2014.002](https://www.nfe.fazenda.gov.br/portal/exibirArquivo.aspx?conteudo=VgPuhtPApQo=), nela tem todas as especificações desse serviço.

**Reply by MikeThmz** [reply]
Perfeito @lucashpmelo , foi justamente o que você falou. Novos documentos começaram a aparecer. Seu componente está funcionando perfeitamente, super simples e eficiente. Obrigado!
Uma outra dúvida, com a consulta foi possível obter todos os documentos emitidos contra um destinatário, porém neste webservice recebemos os documentos resumidos. Você sabe dizer se há um outro webservice onde podemos obter o XML completo com base na chave da NFe que acabamos de obter? 
Obrigado novamente.

**Reply by lucashpmelo** [reply]
Quando o autor que está consultado é o destinatário da NF-e esse serviço sempre irá retornar primeiro na consulta o XML resumido da nota. Para conseguir obter o XML completo é necessário realizar a emissão do Evento de Manifestação do Destinatário que é o MD-e. Você pode fazer isso utilizando essa biblioteca, só consultar a parte de [Manifestação do Destinatário](https://github.com/lucashpmelo/node-mde#manifesta%C3%A7%C3%A3o-do-destinat%C3%A1rio) no [README.md](https://github.com/lucashpmelo/node-mde#readme).

Depois de manifestar as notas resumidas você pode continuar com a consulta pelo ultimo NSU, que a Sefaz irá gerar novos NSUs agora com o XML completo das NF-e.

Para entender o que cada evento significa é os prazos para emissão dos mesmos, você deve consultar a [Nota Técnica 2020.001](https://www.nfe.fazenda.gov.br/portal/exibirArquivo.aspx?conteudo=nPKQWFVzvQo=).

**Reply by MikeThmz** [reply]
Mais uma vez obrigado Lucas. Tudo funciona exatamente conforme sua explicação. Os componentes da sua bibliboteca funcionaram perfeitamente.
